Fixes #2

hecksadecimal opened the pull request:
Changes currPopup to JPopupMenu and implements with the canvas click check to hide the menu when you click elsewhere. Additionally, currPopup is set to Heavyweight mode and added to the Canvas' parent to sidestep some heavyweight/lightweight incompatibility issues.

monster860 commented on src/monster860/fastdmm/FastDMM.java:
```diff
@@ -482,27 +482,33 @@ else if(dwheel < 0)
+					currPopup.setVisible(false);
+				} catch (NullPointerException e) {
+					// Doesn't exist. Oh well.
+				}
```
Please do this properly. like this:

```
if(currPopup != null {
    currPopup.setVisible(false);
    continue;
}
```

hecksadecimal replied:
That changes the behavior. Doing this causes it to disappear correctly the first time, but then it no longer creates the context menu on right click.

monster860 replied:
Put this before the if statement

```
if(currPopup != null && !currPopup.isVisible())
    currPopup = null;
```

And inside, between the setvisible and continue call put `currPopup = null`

monster860 commented on src/monster860/fastdmm/FastDMM.java:
```diff
@@ -482,27 +482,33 @@ else if(dwheel < 0)
 							menu.add(item);
 						}
-						canvas.add(currPopup);
+						canvas.getParent().add(currPopup);
```
Have you tested this to make sure the right click menu actually shows up in the correct place?

hecksadecimal replied:
I have, yes. It behaves as expected.

monster860 commented:
Please do something about the 

`Merge remote-tracking branch 'refs/remotes/monster860/master'`

commits

hecksadecimal commented:
How do you propose I do that? It happens whenever I pull from this repo to get the latest changes.

monster860 commented:
How to fix having these fucked up merge commits on your `master` branch:

```
git fetch monster860
git checkout master
git reset --hard HEAD~<number of merge commits>  (If you have 4 merge commits do git reset --hard HEAD~4
git merge monster860/master --ff-only
git push --force origin master
```

How to fix having these fucked up merge commis on your `Fix-Context` branch:
(Do this after fixing it on your master branch)

```
git checkout
git reset --soft HEAD~
git stash
git reset --hard HEAD~2
git merge master
git stash apply
git push --force origin Fix-Context
```

Do not use the `Update from monster860/master` button in Github. It sucks balls.

Instead, do this in the command line whenever you want to update

```
git fetch monster860
git checkout master
git merge monster860/master --ff-only
```

monster860 commented:
Oh yeah, I forgot to mention you need to actually commit your changes and then you can reopen _this_ pull request.

hecksadecimal commented:
Alrighty. I'm trying your code change suggestions right now.

hecksadecimal committed:
Fixes buggy context menu behavior

PopupMenu behaved inproperly in some cases. It would cause the window to
lock up. It has been substituted with Swing's JPopupMenu. To get around
the fact that Swing uses lightweight components, and the canvas is a
Heavyweight component, currPopup is forced into Heavyweight mode. This
ensures that the context menu is always drawn above the canvas. Custom
checks had to be made to close the context menu when you click out of
it, and attaching it to the parent container of canvas ensures
compatibility.

hecksadecimal commented:
Applied and tested the changes. Behaves as expected.
